### PR TITLE
101-azurefirewall-hub-spoke-demo

### PR DESCRIPTION
**Pull request by mddazure**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by mddazure** [comment]
Review comments from previous pull request incorporated

File: 101-azurefirewall-hub-spoke-demo/metadata.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "type": "QuickStart",
+  "itemDisplayName": "101-azurefirewall-hub-spoke-demo",
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/metadata.json]: Use a friendly name here, e.g. "Azure Firewall Hub & Spoke Demo"

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,1442 @@
...
+        "publicIPAddresses_azfw_on_prem_ip_name": "azfw-on-prem-ip",
+        "publicIPAddresses_azfw_hub_gw_pub_ip_name": "azfw-hub-gw-pub-ip",
+        "publicIPAddresses_azfw_on_prem_gw_pub_ip_name": "azfw-on-prem-gw-pub-ip",
+    
+        "routeTables_azfw_all_to_fw_name": "azfw-all-to-fw",
+        "routeTables_azfw_spokes_via_fw_name": "azfw-spokes-via-fw",
+    
+        "virtualNetworkGateways_azfw_hub_gw_name": "azfw-hub-gw",
+        "virtualNetworkGateways_azfw_on_prem_gw_name": "azfw-on-prem-gw",
+        "virtualNetworkGateways_shared_key":"[uniquestring(deployment().name)]"
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: make this idempotent (e.g. uniqueString(resourceGroup().id))

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,1442 @@
...
+    
+        "routeTables_azfw_all_to_fw_name": "azfw-all-to-fw",
+        "routeTables_azfw_spokes_via_fw_name": "azfw-spokes-via-fw",
+    
+        "virtualNetworkGateways_azfw_hub_gw_name": "azfw-hub-gw",
+        "virtualNetworkGateways_azfw_on_prem_gw_name": "azfw-on-prem-gw",
+        "virtualNetworkGateways_shared_key":"[uniquestring(deployment().name)]"    
+    },
+    "resources": [
+        {
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: use a copy loop with array variables for the publicIps

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,1442 @@
...
+                            "privateIPAddressVersion": "IPv4"
+                        }
+                    }
+                ],
+                "enableAcceleratedNetworking": true,
+                "enableIPForwarding": false
+            }
+        },
+        {
+            "type": "Microsoft.Network/routeTables/routes",
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: same as above, for these duplicate resources, use copy loops

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,1442 @@
...
+                "[resourceId('Microsoft.Network/routeTables', variables('routeTables_azfw_spokes_via_fw_name'))]"
+            ],
+            "properties": {
+                "addressPrefix": "192.168.1.0/24",
+                "nextHopType": "VirtualAppliance",
+                "nextHopIpAddress": "10.100.0.132"
+            }
+        },
+        {
+            "type": "Microsoft.Network/virtualNetworks/subnets",
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: use a copy loop

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,1442 @@
...
+                        "blob": {
+                            "enabled": true
+                        }
+                    },
+                    "keySource": "Microsoft.Storage"
+                }
+            }
+        },
+        {
+            "type": "Microsoft.Network/networkInterfaces",
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: ok, won't comment on the rest, but use copy loops for the resources below where you have multiple, similar resource definitions.

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**